numcodecs/zarr3.py

```diff
@@ -56,6 +58,23 @@
 CODEC_PREFIX = "numcodecs."
 
 
+def _from_zarr_dtype(dtype: Any) -> np.dtype:
```

Comment:
Is the input type here really `Any`?

Reply:
no, but see https://github.com/zarr-developers/numcodecs/pull/766#discussion_r2270237827, it's not easy to annotate this type properly. And IMO we don't really need type safety for these functions right now, because they are very narrowly used.

Reply:
Fair enough.